Issue #407: Migrate to arrow 1.0.0

veganjay opened the pull request:
The arrow library moved to 1.0.0 on February 24, 2021 and with it came breaking changes.
This pull request addresses those changes per the [arrow migration guide](https://github.com/arrow-py/arrow/issues/832).

jmaupetit commented on requirements.txt:
```diff
@@ -1,4 +1,4 @@
-arrow>=0.15.6
+arrow==1.0.0
```
Let's be optimistic!

```suggestion
arrow>=1.0.0
```

jmaupetit replied:
I hope this won't have a catastrophic impact for people installing Watson in their global python distribution...

jmaupetit reviewed (commented):
Looks like you'll have to also fix the tests... Thanks! :pray:

veganjay commented:
> Looks like you'll have to also fix the tests... Thanks!

I took a quick look and there is an issue with timezone localization, making "add --from" and "add --to" work incorrectly.  I need to investigate further.

veganjay commented:
I fixed the unit test cases and also another issue with the timezone.  The proper way to change the timezone of an arrow date is now:

    date = date.replace(tzinfo=tz.tzlocal())

The travis build failed because arrow 1.0.0 is not available for python 3.5.
Given that [Python 3.5 is at end of life](https://www.python.org/downloads/release/python-3510/), it may be worthwhile considering ending support with updating setup.py and .travis.yml accordingly.

jmaupetit commented:
Thank you so much @veganjay :pray: 

>  Given that Python 3.5 is at end of life, it may be worthwhile considering ending support with updating setup.py and .travis.yml accordingly.

Agreed! Can you add consequent changes in this PR & fill the CHANGELOG to mention arrow major release upgrade + dropped python 3.5 support? Thx :heart:

veganjay commented:
Removed support for 3.5 and updated the CHANGELOG.

By the way, I installed *td-watson* with [pipx](https://github.com/pipxproject/pipx):

    pipx install td-watson

This way the arrow library and other dependencies are installed in their own isolated environment.  This is helpful if working on various other virtualenvs during the day.

jmaupetit reviewed (approved):
Thank you @veganjay :pray: :heart: 

![](https://media1.giphy.com/media/d2Z9QYzA2aidiWn6/giphy.gif)
